cassette.go

```diff
@@ -229,34 +236,52 @@ func (k7 *cassette) replayResponse(trackNumber int, req *http.Request) *http.Res
 
 // saveCassette writes a cassette to file.
 func (k7 *cassette) save() error {
-	// marshal
 	data, err := json.Marshal(k7)
 	if err != nil {
 		return err
 	}
 
-	// transform properties known to fail on Unmarshal
-	data, err = transformInterfacesInJSON(data)
+	tData, err := transformInterfacesInJSON(data)
 	if err != nil {
 		return err
 	}
 
 	// beautify JSON (now that the JSON text has been transformed)
 	var iData bytes.Buffer
-
-	if err := json.Indent(&iData, data, "", "  "); err != nil {
+	if err := json.Indent(&iData, tData, "", "  "); err != nil {
```

Comment:
In general, I would consider "beautifying" data that is compressed anyway as a waste of CPU and space, which is why I disabled it for compressed images.  It will probably not have a huge impact after compression, but still.

Reply:
It's a fair point, but sometimes I edit the cassettes manually to generate scenarios (typically "unhappy path") which would otherwise be difficult to create with a live system (where I can't force unsual behaviour or failure).